julia/deps/build_local.jl

```diff
@@ -0,0 +1,68 @@
+using Pkg
+# save the currently active project path to use later when saving preferences
+project_path = dirname(Pkg.project().path)
+
+# Install build time depenedencies
+Pkg.activate(@__DIR__)
+Pkg.instantiate()
+
+
+using Scratch, Preferences, CMake_jll
+
+Metatensor = Base.UUID("afba2530-6023-4bf7-b247-2c594027d0bd")
+
+# get scratch directories
+build_dir = get_scratch!(Metatensor, "build")
+isdir(build_dir) && rm(build_dir; recursive=true)
+
+install_dir = get_scratch!(Metatensor, "usr")
+isdir(install_dir) && rm(install_dir; recursive=true)
+
+source_dir = realpath(joinpath(@__DIR__, "..", "..", "metatensor-core"))
+isdir(source_dir) || error("could not find metatensore-core sources")
+
+# build and install
+@info "Building metatensor from sources" source_dir build_dir install_dir
+cmake() do cmake_path
+    cmake_opts = [
+        "-DBUILD_SHARED_LIBS=ON",
+        "-DMETATENSOR_INSTALL_BOTH_STATIC_SHARED=OFF",
+        "-DCMAKE_INSTALL_PREFIX=$(install_dir)",
+    ]
+    cmd = `$cmake_path $cmake_opts -B$(build_dir) -S$(source_dir)`
+    @info "Configuring CMake" cmd
+    run(cmd)
+
+    cmd = `$cmake_path --build $(build_dir) --target install`
+    @info "Building with CMake" cmd
+    run(cmd)
+end
+
+possible_paths = [
+    joinpath(install_dir, "lib", "libmetatensor.so"),
+    joinpath(install_dir, "lib", "libmetatensor.dylib"),
+    joinpath(install_dir, "bin", "metatensor.dll"),
+]
+
+lib_path = missing
+for path in possible_paths
+    if isfile(path)
+        global lib_path
+        lib_path = path
+        break
+    end
+end
+
+if ismissing(lib_path)
+    error("could not find metatensor library in install directory")
+end
+
+# Save the path to the freshly built library in preferences, it will be checked
+# when loading Metatensor.
+@info "Setting preference to use local build in $(joinpath(project_path, "LocalPreferences.toml"))"
+set_preferences!(
```

Comment:
interesting the help page does not show all signatures
```
help?>  set_preferences!
search:        

  set_preferences!(uuid_or_module, prefs::Pair{String,Any}...; export_prefs=false,
                   active_project_only=true, force=false)                                                                                                                            
```
but it internally actually knows it
```
julia> set_preferences!(
           "Metatensor",
           "libmetatensor" => lib_path;
           force=true
       )
ERROR: MethodError: no method matching set_preferences!(::String, ::Pair{String, String}; force::Bool)

Closest candidates are:
  set_preferences!(::Module, ::Pair{String}...; kwargs...)
   @ Preferences /opt/anaconda3/envs/metatensor-dev/share/julia/packages/Preferences/TTEAN/src/Preferences.jl:285
  set_preferences!(::Base.UUID, ::Pair{String}...; export_prefs, active_project_only, kwargs...)
   @ Preferences /opt/anaconda3/envs/metatensor-dev/share/julia/packages/Preferences/TTEAN/src/Preferences.jl:205
  set_preferences!(::String, ::String, Pair{String}...; force)
   @ Preferences /opt/anaconda3/envs/metatensor-dev/share/julia/packages/Preferences/TTEAN/src/Preferences.jl:115

Stacktrace:
 [1] top-level scope
   @ REPL[42]:1
```

Reply:
Yes, the help page is potentially user generated, but the method dispatch uses the full compiler